### PR TITLE
comment hightlight id support

### PR DESCRIPTION
**Pull request by DZGoldman**
includes adding a comment id as a data-attribute along with background color, removing all comment highlights from squire content (for comment resolution), and logic for when to merge adjacent comment highlights

### CODE REVIEW

File: source/Clean.js [hunk 1]
```diff
@@ -454,6 +454,29 @@ var cleanTree = function cleanTree ( node, preserveWS, nestedNodeTracker ) {
     return node;
 };
 
+
+var removeCommentHighlights = function removeCommentHighlights (node, commentId){
```
Comment (DZGoldman) [inline source/Clean.js:458]: this uses a similar approach to the cleanTree function

File: source/Node.js [hunk 2]
```diff
@@ -177,12 +177,38 @@ function findNextTextOrNotEditable (root, node, options){
...
+
+function preventCommentHighlightMerge(n1, n2) {
+    var n1HighlightId = commentHighlightId(n1)
+    var n2HighlightId = commentHighlightId(n2)
+    // if neither are highlights, don't prevent
+    if (!n1HighlightId && !n1HighlightId){
+        return false
+    }
+    // if exactly one is a highlight, prevent merge
+    else if (!n1HighlightId || !n1HighlightId){
```
Comment (krzysiek1507) [inline source/Node.js:192]: When you return from `if` block, you don't need to write `else if` so another `if` sufficient.

Comment (DZGoldman) [reply source/Node.js:192]: yeah, just a stylistic thing

**Review by krzysiek1507 — APPROVED** [review]
It looks OK for me. @jayshenk what do you think?

**Review by jayshenk — APPROVED** [review]

**Pull request merged**